src/components/TopicLoader/TopicLoader.js

```diff
@@ -83,13 +85,23 @@ class TopicLoader extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { activeTopic } = this.props;
+    const {
+      activeTopic,
+      dispatchSetClickedFeatureInfo,
+      searchService,
+    } = this.props;
 
     if (activeTopic !== prevProps.activeTopic) {
       if (activeTopic.linkUrl) {
         TopicLoader.openLinkTopic(activeTopic);
       }
       this.updateLayers(activeTopic.layers);
+
+      searchService.setSearches(activeTopic.searches);
+      searchService.setSearchProps({
+        activeTopic,
+        dispatchSetClickedFeatureInfo,
```

Comment:
since searchs meant to be generic acroos applications, I would not put dispatch function in properties. may be an 

`onItemSelect: (featureInfo) => {
  dispatchSetClickedFeatureInfo(featureInfo)
}`  
will be more appropriate.

Reply:
`dispatchSetClickedFeatureInfo` will only be used by `HandicapStationFinder`, so a generic `onItemSelect` callback for all searches will not work in this case. Also `setSearchProps` will  only be called by custom apps and those custom props only be consumed by custom SearchClasses specific to the app (`HandicapStationFinder` should not be moved to `react-transit`). It looks a little awkward, but it's the best way I found for this problem.

Reply:
Ok I see , i can live with that